docs/contributing.md

```diff
@@ -0,0 +1,115 @@
+# Contributing to the monorepo
+
+## Getting started
+
+- Install [Node.js](https://nodejs.org) version 14.
+  - If you're using [NVM](https://github.com/creationix/nvm#installation) (recommended), `nvm use` will ensure that the right version is installed.
+- Install [Yarn v3](https://yarnpkg.com/getting-started/install).
+- Run `yarn install` to install dependencies and run any required post-install scripts.
+- Run `yarn simple-git-hooks` to add a [Git hook](https://github.com/toplenboren/simple-git-hooks#what-is-a-git-hook) to your local development environment which will ensure that all files pass linting before you push a branch.
+
+## Testing
+
+### Running unit tests
+
+- Run `yarn workspace <workspaceName> run test` to run all tests for a package.
+- Run `yarn workspace <workspaceName> run jest --no-coverage <file>` to run a test file within the context of a package.
+- Run `yarn test` to run tests for all packages.
+
+> **Note**  
+> `workspaceName` here is the `name` field within a package's `package.json`, e.g., `@metamask/announcement-controller`, not the directory where it is located, e.g., `packages/announcement-controller`.
+
+### Integrating with other projects
+
+When developing changes to packages within this repository that a different project depends upon, you may wish to load those changes into the project and test them locally or in CI before publishing proper releases of those packages. To solve that problem, this repository provides a mechanism to publish "preview" versions of packages to GitHub Package Registry. These versions can then be used in the project like any other version, provided the project is configured to use that registry.
+
+#### As a MetaMask contributor
+
+If you're a MetaMask engineer, you can create these preview versions via draft pull requests:
+
+1. Navigate to your settings within GitHub and [create a classic access token](https://docs.github.com/en/authentication/keeping-your-account-and-data-secure/creating-a-personal-access-token#creating-a-personal-access-token-classic). Make sure to give this token the `packages:read` scope.
+2. Switch to your project locally and add a `.npmrc` file with the following content, filling in the appropriate areas:
+   ```
+   @metamask:registry=https://npm.pkg.github.com
+   //npm.pkg.github.com/:_authToken=<your personal access token>
+   ```
+   Make sure not to commit this file.
+3. Go to GitHub and open up a pull request for this repository (for the extension or mobile app, this needs to be a draft PR), then post a comment on the PR with the text `@metamask-bot preview-build`. (This triggers the `publish-preview` GitHub action.)
```

Comment:
There isn't any requirement for this command to be used in draft PRs
```suggestion
3. Go to GitHub and open up a pull request for this repository, then post a comment on the PR with the text `@metamask-bot preview-build`. (This triggers the `publish-preview` GitHub action.)
```

Reply:
Ah, sorry. Right. I don't know why I got that mixed up.

Reply:
Fixed in [bae8925](https://github.com/MetaMask/controllers/pull/993/commits/bae89258a3e9936fa0f6b3c4214a4f011cadd749).